### PR TITLE
More flexible scoring module by allowing the users to put more info i…

### PR DESCRIPTION
**Pull request by XianXing**
…nto the scored items.

This is done most by adding a few optional fields to ```ScoredItem```, and adapting the scoring driver to take advantage of that.

Also added a few tests to cover the changes made as well as deleting tests that are no longer needed.

This PR addresses issue #71.

### CODE REVIEW

File: photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/ScoredItem.scala [hunk 1]
```diff
@@ -14,20 +14,16 @@
  */
 package com.linkedin.photon.ml.cli.game.scoring
 
+import scala.collection.Map
+
 /**
   * A compact representation of the scored item
   * @param uid A uid that can be used to uniquely characterize the scored item
   * @param predictionScore Score predicted by the model
   */
```
Comment (joshvfleming) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/ScoredItem.scala]: The param documentation is out-of-date now.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/avro/data/ScoreProcessingUtils.scala [hunk 2]
```diff
@@ -51,9 +57,19 @@ object ScoreProcessingUtils {
...
+      val metaDataMap = collection.mutable.Map(ids.toMap[CharSequence, CharSequence].toSeq: _*).asJava
+      val builder = ScoringResultAvro.newBuilder()
+      builder.setPredictionScore(predictionScore)
+      builder.setModelId(modelId)
+      if (uidOpt.isDefined) {
+        builder.setUid(uidOpt.get)
+      }
+      if (labelOpt.isDefined) {
+        builder.setLabel(labelOpt.get)
+      }
```
Comment (ashelkovnykov) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/avro/data/ScoreProcessingUtils.scala]: I believe the recommended syntax for these Option checks would be:
```
uidOpt.foreach(builder.setUid(_))
labelOpt.foreach(builder.setLabel(_))
```
http://blog.tmorris.net/posts/scalaoption-cheat-sheet/

Comment (XianXing) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/avro/data/ScoreProcessingUtils.scala]: Good point.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/ScoredItem.scala (hunk 1 above)
Comment (XianXing) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/ScoredItem.scala]: Thanks for pointing it out.

**Pull request merged**